src/System.Net.Http/src/System/Net/Http/Managed/HttpConnectionPool.cs

```diff
@@ -21,9 +21,16 @@ public HttpConnectionPool()
 
         public HttpConnection GetConnection()
         {
-            HttpConnection connection;
-            if (_idleConnections.TryTake(out connection))
+            while (_idleConnections.TryTake(out HttpConnection connection))
             {
+                if (connection.InitialReadCompleted)
+                {
+                    // We got a connection, but it was either already closed by the server or the
```

Comment:
It would be nice if we took the connection out of the pool as soon as the read completed.  That said, I don't think we necessarily need to do this as part of this PR, but it would be nice to add a comment that we should revisit.